tests/tensorflow/test_printing.py

```diff
@@ -79,3 +79,67 @@ def test_unknown_shape():
     expected_out = 'Tensor(Placeholder):0,\tshape=Unknown\t"A:0"\n'
 
     assert std_out.getvalue() == expected_out.lstrip()
+
+
+@pytest.mark.usefixtures("run_with_tensorflow")
+@run_in_graph_mode
+def test_shape_index():
+    """Make sure graph indexing functions as expected."""
+
+    A = tf.compat.v1.placeholder('float', name='A',
+                                 shape=tf.TensorShape([None, None]))
+    x = tf.compat.v1.placeholder('float', name='x',
+                                 shape=tf.TensorShape([None, 1]))
+    y = tf.multiply(1.0, x, name='y')
+
+    z = tf.matmul(A, tf.add(y, y, name='x_p_y'), name='A_dot')
+
+
+    range_out = io.StringIO()
+    with redirect_stdout(range_out):        
+        tf_dprint(z, depth_lower=3, depth_upper=6)
```

Comment:
Use a graph and/or range that truncates a valid cross-section/slice of the graph (i.e. doesn't go past the beginning or end of a graph).

Reply:
Updated in latest commit.